### DISCUSSION TITLE
How to dynamically add fields to a deeply nested object?

### DISCUSSION THREAD

**Discussion by ZipBrandon** [react-hook-form/react-hook-form#6470] [Q&A]
I'm working with an array of objects and I am trying to `register` a new field on a heavily nested field `props`.  When I register it (`pagesConfiguration.pages.16.props.imageUrl`) I get `index.esm.js:156 Uncaught TypeError: Cannot add property imageUrl, object is not extensible`. Here's what it that `pagesConfiguration.pages.16` currently is:
```
actions: {onMount: Array(1)}
component: "VideoPage"
id: "id here"
props: {}
sections: [{…}]
```

How can I dynamically add fields to this deeply nested object?

**Comment by JamesWong1999** [comment]
can you share a codesandbox? also, file object is always tricky with security rules.

**Reply by ZipBrandon** [reply]
Hello and thank you.  I attempted to create a CodeSandbox but wasn't able to reproduce the problem and that led to a discovery!  My `defaultValues` comes from the result of an Apollo GraphQL operation.  I was putting that directly into the form.  I surmise that didn't work since it wasn't all primitives and referenced objects that were under the control of Apollo.  My fix was to `cloneDeep` the result from GraphQL operation and that has seemed to remedy it.
